index.js

```diff
@@ -77,11 +77,13 @@ const requestAutoSignin = async () => {
         subtype: "auto-signin-signature",
       });
       if (error) {
-        window.postMessage({ type: "select-auto-signin", requestId }, "*");
-        pubsub.subscribe(requestId, (_event, data) => {
-          resolve(data);
-          pubsub.unsubscribe(requestId);
-        });
+        if (error.code === 404) {
```

Comment:
What should this do if it isn't a 404?

Reply:
For this particular request, an error with 404 is thrown by browser service worker not by the server. It means no auto signin found for current domain and user needs to select one by opening the extension.

Reply:
> For this particular request, an error with 404 is thrown by browser service worker not by the server. It means no auto signin found for current domain and user needs to select one by opening the extension.

That seems reasonable. Should there be a check for other others, in case the service worker is changed and starts throwing an additional error?

Reply:
> > For this particular request, an error with 404 is thrown by browser service worker not by the server. It means no auto signin found for current domain and user needs to select one by opening the extension.
> 
> That seems reasonable. Should there be a check for other others, in case the service worker is changed and starts throwing an additional error?

You are right, great catch 👏 . I think it would reject error in that case to throw it towards webapp. I'll make a fix shortly.